tests/src/test/scala/org/apache/openwhisk/core/containerpool/test/ContainerPoolTests.scala

```diff
@@ -753,14 +757,102 @@ class ContainerPoolTests
+    val pool =
+      system.actorOf(
+        ContainerPool
+          .props(
+            factory,
+            poolConfig(MemoryLimit.STD_MEMORY * 4),
+            feed.ref,
+            List(PrewarmingConfig(2, exec, memoryLimit, reactive))))
+    containers(0).expectMsg(Start(exec, memoryLimit, Some(ttl)))
+    containers(1).expectMsg(Start(exec, memoryLimit, Some(ttl)))
+    containers(0).send(pool, NeedWork(preWarmedData(exec.kind, expires = Some(ttl.fromNow))))
+    containers(1).send(pool, NeedWork(preWarmedData(exec.kind, expires = Some(ttl.fromNow))))
+    stream.reset()
+
+    // Make sure prewarmed containers can be deleted from prewarmedPool due to unused
+    Thread.sleep(3.seconds.toMillis)
+
+    pool ! AdjustPrewarmedContainer
+    containers(0).expectMsg(Remove)
+    containers(1).expectMsg(Remove)
+    containers(0).send(pool, ContainerRemoved(false))
+    containers(1).send(pool, ContainerRemoved(false))
+
+    // Make sure prewarmed containers are deleted
+    Thread.sleep(2.seconds.toMillis)
+
+    stream.toString should include("prewarmed container is deleted")
+    stream.reset()
+  }
+
+  it should "supplement prewarmed container when doesn't have enough container to handle activation" in {
+    val (containers, factory) = testContainers(6)
+    val feed = TestProbe()
+
+    val deadline: Option[Deadline] = Some(ttl.fromNow)
+    val reactive: Option[ReactivePrewarmingConfig] = Some(ReactivePrewarmingConfig(0, 2, ttl, threshold, increment))
+    val pool =
+      system.actorOf(
+        ContainerPool
+          .props(
+            factory,
+            poolConfig(MemoryLimit.STD_MEMORY * 4),
+            feed.ref,
+            List(PrewarmingConfig(2, exec, memoryLimit, reactive))))
+    containers(0).expectMsg(Start(exec, memoryLimit, Some(ttl)))
+    containers(1).expectMsg(Start(exec, memoryLimit, Some(ttl)))
+    containers(0).send(pool, NeedWork(preWarmedData(exec.kind, expires = deadline)))
+    containers(1).send(pool, NeedWork(preWarmedData(exec.kind, expires = deadline)))
+    stream.reset()
+
+    // Make sure prewarmed containers can be deleted from prewarmedPool due to unused
+    Thread.sleep(3.seconds.toMillis)
+
+    pool ! AdjustPrewarmedContainer
+    containers(0).expectMsg(Remove)
+    containers(1).expectMsg(Remove)
+    containers(0).send(pool, ContainerRemoved(false))
+    containers(1).send(pool, ContainerRemoved(false))
+
+    val action = ExecutableWhiskAction(
+      EntityPath("actionSpace"),
+      EntityName("actionName"),
+      exec,
+      limits = ActionLimits(memory = MemoryLimit(memoryLimit)))
+    val run = createRunMessage(action, invocationNamespace)
+
+    // Make sure prewarmed containers are deleted
+    Thread.sleep(2.seconds.toMillis)
+
+    pool ! run
+    pool ! run
+    containers(2).expectMsg(run)
+    containers(3).expectMsg(run)
+
+    // Make sure cold start in previous 1 minute can be counted
+    Thread.sleep(60.seconds.toMillis)
```

Comment:
the schedule period should be configurable so this test doesn't need to wait 60s

Reply:
Already make schedule period configurable.